libraries/AP_OpticalFlow/OpticalFlow.cpp

```diff
@@ -120,10 +118,66 @@ void OpticalFlow::init(void)
 
 void OpticalFlow::update(void)
 {
+    // exit immediately if not enabled
+    if (!enabled()) {
+        return;
+    }
+
     if (backend != nullptr) {
         backend->update();
     }
+
     // only healthy if the data is less than 0.5s old
     _flags.healthy = (AP_HAL::millis() - _last_update_ms < 500);
 }
 
+void OpticalFlow::update_state(const OpticalFlow_state &state)
+{
+    _state = state;
+    _last_update_ms = AP_HAL::millis();
+
+    // write to log and send to EKF if new data has arrived
+    AP::ahrs_navekf().writeOptFlowMeas(quality(),
+                                       _state.flowRate,
+                                       _state.bodyRate,
+                                       _last_update_ms,
+                                       get_pos_offset());
+    Log_Write_Optflow();
+}
+
+void OpticalFlow::Log_Write_Optflow()
+{
+    DataFlash_Class *instance = DataFlash_Class::instance();
+    if (instance == nullptr) {
+        return;
+    }
+    if (_log_bit == (uint32_t)-1 ||
```

Comment:
Doesn't this cause plane to always log while disarmed regardless of what the user wants since the bit is -1?

Would it be reasonable to extend `DataFlash_Class::should_log(const uint32_t mask)` to accept/manage the special case of -1? (Alternatively you could just pass in -1 anyways and that would mean if any logging is enabled that we would log this message (which seems appropriate since you are trying to always log anyways)).

Reply:
So I've changed behaviour here for Copter and Sub but not Plane.  Previously they all would have ignored the arm and download states etc etc.  Now only Plane does....

This bit bothers me a lot, actually.  AP_InertialSensor_Backend also takes a `-1` bit for logging - but that value over there means "never log raw imu data".  This makes it problematic to handle this in the DataFlash library.

And what's "Resolve conversation"?! :-)

Reply:
Means we can dismiss a point I raised as "acceptable" without you having to change code, and makes it easier for future reviewers. :)